lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -446,40 +446,22 @@ def _get(
                    }             }           }               }
                   }             }           }               }
         """
-        mtable = aliased(Metadata)
-        stable = aliased(Metadata)
-        gtable = aliased(Metadata)
-        utable = aliased(Metadata)
         aliases = {
-            Metadata.METALOG: mtable,
-            Metadata.SERVER: stable,
-            Metadata.GLOBAL: gtable,
-            Metadata.USER: utable,
+            Metadata.METALOG: aliased(Metadata),
+            Metadata.SERVER: aliased(Metadata),
+            Metadata.GLOBAL: aliased(Metadata),
+            Metadata.USER: aliased(Metadata),
         }
-        query = (
-            Database.db_session.query(Dataset)
-            .outerjoin(
-                mtable,
-                and_(mtable.dataset_ref == Dataset.id, mtable.key == Metadata.METALOG),
-            )
-            .outerjoin(
-                stable,
-                and_(stable.dataset_ref == Dataset.id, stable.key == Metadata.SERVER),
-            )
-            .outerjoin(
-                gtable,
-                and_(gtable.dataset_ref == Dataset.id, gtable.key == Metadata.GLOBAL),
-            )
-        )
-        if auth_id:
-            query = query.outerjoin(
-                utable,
-                and_(
-                    utable.dataset_ref == Dataset.id,
-                    utable.key == Metadata.USER,
-                    utable.user_id == auth_id,
-                ),
-            )
+        query = Database.db_session.query(Dataset)
+        for key, table in aliases.items():
+            terms = [table.dataset_ref == Dataset.id, table.key == key]
+            if key == Metadata.USER:
+                if auth_id:
+                    terms.append(table.user_id == auth_id)
+                else:
+                    continue
+            current_app.logger.info("Adding JOIN {}", key)
```

Comment:
Are you sure you want an `INFO`-level log for this in production code?

Reply:
Sigh; no, that was a debug line I missed. 😦